pkg/cluster/private_registry_test.go

```diff
@@ -110,6 +110,81 @@ func TestGeneratePrivateRegistryDockerConfig(t *testing.T) {
 				},
 			},
 		},
+		{
+			name:           "v2prov private registry with upstream registry",
+			expectedUrl:    "0123456789abcdef.dkr.ecr.us-east-1.amazonaws.com",
+			expectedConfig: base64.StdEncoding.EncodeToString([]byte(`{"auths":{"0123456789abcdef.dkr.ecr.us-east-1.amazonaws.com":{"username":"testuser","password":"password","auth":"dGVzdHVzZXI6cGFzc3dvcmQ="}}}`)),
+			expectedError:  "",
+			cluster: &v3.Cluster{
+				Spec: v3.ClusterSpec{
+					ClusterSpecBase: v3.ClusterSpecBase{
+						RancherKubernetesEngineConfig: &rketypes.RancherKubernetesEngineConfig{
+							PrivateRegistries: []rketypes.PrivateRegistry{{
+								URL: "upstream-registry.com",
+							}},
+						},
+						ClusterSecrets: v3.ClusterSecrets{
+							PrivateRegistrySecret: "test-secret",
+							PrivateRegistryURL:    "0123456789abcdef.dkr.ecr.us-east-1.amazonaws.com",
+						},
+					},
+					FleetWorkspaceName: "fleet-default",
+				},
+			},
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "fleet-default",
+						Name:      "test-secret",
+					},
+					Data: map[string][]byte{
+						"username": []byte("testuser"),
+						"password": []byte("password"),
+					},
+				},
+			},
+		},
+		{
+			name:           "v2prov public registry with upstream registry",
+			expectedUrl:    "0123456789abcdef.dkr.ecr.us-east-1.amazonaws.com",
+			expectedConfig: "",
+			expectedError:  "",
+			// cluster.Spec.RancherKubernetesEngineConfig.PrivateRegistries
+			cluster: &v3.Cluster{
+				Spec: v3.ClusterSpec{
+					ClusterSpecBase: v3.ClusterSpecBase{
+						RancherKubernetesEngineConfig: &rketypes.RancherKubernetesEngineConfig{
+							PrivateRegistries: []rketypes.PrivateRegistry{{
+								URL: "upstream-registry.com",
+							}},
+						},
+						ClusterSecrets: v3.ClusterSecrets{
+							PrivateRegistryURL: "0123456789abcdef.dkr.ecr.us-east-1.amazonaws.com",
+						},
+					},
+					FleetWorkspaceName: "fleet-default",
+				},
+			},
+		},
+		{
+			name:           "Upstrem registry and no downstream one",
```

Comment:
Could we rename this to `global system default registry and no cluster default registry`? It can get pretty confusing if we don't stick to the current terminology